### PR TITLE
split system.network.connections into TCP and UDP metrics

### PR DESCRIPTION
**Pull request by codeboten**
As a follow up to https://github.com/open-telemetry/opentelemetry-specification/pull/2663, since UDP has no state attribute, I propose UDP and TCP connection metrics should be split into protocol specific metric for each supported protocol.

### CODE REVIEW

**Review by tigrannajaryan — DISMISSED** [review]
We need to review/merge [this PR](https://github.com/open-telemetry/opentelemetry-specification/pull/2653) before splits can be published in the next release.

I am blocking this to make it clear that this depends on #2653, not because I oppose this PR.

**Comment by tigrannajaryan** [comment]
Please add this change to the schema file 1.13.

File: specification/metrics/semantic_conventions/system-metrics.md [hunk 1]
```diff
@@ -146,9 +146,12 @@ perf counter (similar for Writes)
...
+| system.network.tcp.connections                  | Count of TCP connections                                                                  | {connections} | UpDownCounter                                     | Int64      | device        | (identifier)                                                                                   |
+|                                                 |                                                                                           |               |                                                   |            | state         | SHOULD be one of: close, close_wait, closing, delete, established, fin_wait_1, fin_wait_2, last_ack, listen, syn_recv, syn_sent, time_wait. A stateless protocol MUST NOT set this attribute. |
+| system.network.udp.connections                  | Count of UDP connections                                                                  | {connections} | UpDownCounter                                     | Int64      | device        | (identifier)                                                                                   |
```
Comment (Oberon00) [inline specification/metrics/semantic_conventions/system-metrics.md:154]: Edge case, but what about other connections? Should there be `system.network.other.connections`?

Comment (codeboten) [reply specification/metrics/semantic_conventions/system-metrics.md:154]: i'd expect any other protocol to follow the same pattern of `system.network.<protocol>.connections`

**Comment by codeboten** [comment]
This PR is not stale, I'm waiting on https://github.com/open-telemetry/opentelemetry-specification/pull/2688 to merge before adding the schema to this PR

**Comment by tigrannajaryan** [comment]
> This PR is not stale, I'm waiting on #2688 to merge before adding the schema to this PR

#2688 merged.

**Comment by codeboten** [comment]
@tigrannajaryan schema file added 👍🏻

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**